src/Common/src/System/Linq/LowLevelEnumerable.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+namespace System.Linq
+{
+    internal static class LowLevelEnumerable
+    {
+        public static bool Any<T>(this IEnumerable<T> values)
+        {
+            Debug.Assert(values != null);
+
+            IEnumerator<T> enumerator = values.GetEnumerator();
+            return enumerator.MoveNext();
```

Comment:
This needs to dispose of the enumerator:
```C#
using (var enumerator = values.GetEnumerator())
    return enumerator.MoveNext();
```
Otherwise for things like file system enumerators and others that have clean up to do, it'll get left for finalization or potentially never be done.